file-formats/cfdf/type/zif_aff_cfdf_v1.intf.abap

```diff
@@ -0,0 +1,589 @@
+INTERFACE zif_aff_cfdf_v1
+  PUBLIC.
+
+  TYPES: ty_field_name              TYPE c LENGTH 30.
+  TYPES: ty_component_identifier    TYPE c LENGTH 5.
+  TYPES: ty_business_context        TYPE c LENGTH 30.
+  TYPES: ty_code_value              TYPE c LENGTH 30.
+  TYPES: ty_data_subject_id_type    TYPE c LENGTH 10.
+  TYPES: ty_type                    TYPE c LENGTH 10.
+  TYPES: ty_message_direction       TYPE c LENGTH 3.
+  TYPES: ty_field_control_property  TYPE c LENGTH 1.
+  TYPES: ty_aggregation             TYPE c LENGTH 5.
+
+  TYPES:
+    "! <p class="shorttext">Label and Tooltip</p>
+    "! Label and tooltip
+    BEGIN OF ty_gs_ui_text,
+      "! <p class="shorttext">Translation Language</p>
+      "! Translation language
+      "! $required
+      language      TYPE sy-langu,
+
+      "! <p class="shorttext">Field Label</p>
+      "! Field label
+      "! $required
+      "! $maxLength 40
+      field_label   TYPE c LENGTH 40,
+
+      "! <p class="shorttext">Field Tooltip</p>
+      "! Field tooltip
+      "! $required
+      "! $maxLength 60
+      field_tooltip TYPE c LENGTH 60,
+    END OF ty_gs_ui_text.
+  TYPES: ty_gt_ui_text TYPE STANDARD TABLE OF ty_gs_ui_text WITH KEY language.
+
+  TYPES:
+    "! <p class="shorttext">Extended Business Context Details</p>
+    "! Extended business context details
+    BEGIN OF ty_gs_extended_bus_context,
+      "! <p class="shorttext">Business Context</p>
+      "! Business context
+      "! $required
+      "! $maxLength 30
```

Comment:
Again here: Even if you use an own type (which if of `TYPE c LENGTH 30`), you don't need to specify maxLength again. Our generator also looks into you defined type to get the maximum length.

Reply:
All occurrences of $maxLength were removed...